lib/ansible/modules/cloud/hcloud/hcloud_floating_ip.py

```diff
@@ -0,0 +1,303 @@
+        else:
+            self.module.fail_json(msg="server or home_location is required")
+
+        if self.module.params.get("labels") is not None:
+            params["labels"] = self.module.params.get("labels")
+        if not self.module.check_mode:
+            resp = self.client.floating_ips.create(**params)
+            self.hcloud_floating_ip = resp.floating_ip
+
+        self._mark_as_changed()
+        self._get_floating_ip()
+
+    def _update_floating_ip(self):
+        labels = self.module.params.get("labels")
+        if labels is not None and labels != self.hcloud_floating_ip.labels:
+            if not self.module.check_mode:
+                self.hcloud_floating_ip.update(labels=labels)
+            self._mark_as_changed()
+
+        description = self.module.params.get("description")
+        if description is not None and description != self.hcloud_floating_ip.description:
+            if not self.module.check_mode:
+                self.hcloud_floating_ip.update(description=description)
+            self._mark_as_changed()
+
+        server = self.module.params.get("server")
+        if server is not None:
+            if not self.module.check_mode:
+                if self.module.params.get("force_assign") or self.hcloud_floating_ip.server is None:
+                    self.hcloud_floating_ip.assign(
+                        self.client.servers.get_by_name(self.module.params.get("server"))
+                    )
+                else:
+                    self.module.warn(
+                        "Floating IP is already assigned to server %s. You need to unassign the Floating IP or use force_assign=yes."
+                        % self.hcloud_floating_ip.server.name
+                    )
+            self._mark_as_changed()
+        elif server is None and self.hcloud_floating_ip.server is not None:
+            if not self.module.check_mode:
+                self.hcloud_floating_ip.unassign()
+            self._mark_as_changed()
+
+        self._get_floating_ip()
+
+    def present_floating_ip(self):
+        self._get_floating_ip()
+        if self.hcloud_floating_ip is None:
+            self._create_floating_ip()
+        else:
+            self._update_floating_ip()
+
+    def delete_floating_ip(self):
+        self._get_floating_ip()
+        if self.hcloud_floating_ip is not None:
+            if not self.module.check_mode:
+                self.client.floating_ips.delete(self.hcloud_floating_ip)
+            self._mark_as_changed()
+        self.hcloud_floating_ip = None
+
+    @staticmethod
+    def define_module():
+        return AnsibleModule(
+            argument_spec=dict(
+                id={"type": "int"},
+                name={"type": "str"},
+                description={"type": "str"},
+                server={"type": "str"},
+                home_location={"type": "str"},
+                force_assign={"type": "bool"},
+                type={"choices": ["ipv4", "ipv6"], "required": True},
```

Comment:
I would suggest to default this to `ipv4` or alternatively only for state=present. because it would be weird to delete an e.g. ipv4 by name and specify type=ipv6 right?